config/routes.rb

```diff
@@ -0,0 +1,13 @@
+Rails.application.routes.draw do
+  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+  root 'search#new'
+
+  post '/search', to: 'search#index', as: 'search_create'
+
+  get ':q', to: 'search#index', as: 'search_index'
```

Comment:
This feels odd to me - having two different verbs going to the same action is a little confusing. Could you consolidate these?